src/js/apis/c2c/DocumentService.js

```diff
@@ -7,6 +7,51 @@ DocumentService.prototype.getAll = function (params) {
   return this.api.get('/' + this.documentType + 's', { params });
 };
 
+DocumentService.prototype.fullDownload = function (params, limit, progress) {
+  // will load the ENTIRE list of document. Limited to 2000 docs
+  const this_ = this;
+
+  limit = limit || 2000;
+  params = { ...params }; // clone
+
+  return new Promise((resolve, reject) => {
+    const result = [];
+
+    params.limit = 100;
+
+    const download = function (offset) {
+      params.offset = offset;
+
+      this_
+        .getAll(params)
+        .then((response) => {
```

Comment:
Use destructuring
```suggestion
        .then(({data, length}) => {
```

to simplify code below

Reply:
lenght is not used. I've used it for data though